src/web/src/components/AdvancedSearchDialog.tsx

```diff
@@ -0,0 +1,187 @@
+import useSearchValue from '../hooks/use-search-value';
+import CloseIcon from '@material-ui/icons/Close';
+import { Dialog, DialogContent, DialogTitle, IconButton } from '@material-ui/core';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import React, { useState } from 'react';
+
+interface Props {
+  setOpenDialog: Function;
+  openDialog: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      position: 'absolute',
+      top: '50%',
+      transform: 'translateY(-120%)',
+    },
+    dialogTitle: {
+      display: 'flex',
+      flexDirection: 'column',
+      width: '600px',
+      [theme.breakpoints.down('xs')]: {
+        width: 'calc(100vw - 80px)',
+      },
+      height: '25px',
+      padding: 15,
+      background: '#eeeeee',
+      overflow: 'hidden',
+    },
+    dialogContent: {
+      background: '#eeeeee',
+      height: '300px',
+      color: 'black',
+      fontSize: '1.8rem',
+      '& p': {
+        color: '#999999',
+      },
+    },
+    input: {
+      width: '100%',
+      height: '25px',
+      boxSizing: 'border-box',
+      borderRadius: '20px',
+      padding: '5px 18px',
+      outline: 'none',
+      border: 'none',
+      marginBottom: '15px',
+    },
+    closeIcon: {
+      fontSize: '2.8rem',
+      position: 'absolute',
+      padding: '10px',
+      color: '#999999',
+    },
+    searchButton: {
+      padding: '10px 25px',
+      position: 'absolute',
+      right: '10px',
+      top: '10px',
+      border: 'solid 1px #1ea0f0',
+      borderRadius: '20px',
+      outline: 'none',
+      cursor: 'pointer',
+      color: '#fff',
+      background: '#1ea0f0',
+      transition: 'background .2s',
+      fontWeight: 'bold',
+      '&:hover': {
+        background: '#3da8e8',
+      },
+    },
+    title: {
+      display: 'inline',
+      position: 'absolute',
+      top: 0,
+      marginLeft: '20px',
+      overflow: 'hidden',
+      fontSize: '1.8rem',
+      color: '#999999',
+    },
+    formControl: {
+      padding: 0,
+      margin: 0,
+    },
+    formLabel: {
+      fontSize: '1.8rem',
+      color: 'black',
+    },
+  })
+);
+
+const AdvancedSearchDialog = (props: Props) => {
+  const classes = useStyles();
+  const { text, onTextChange, onFilterChange, onSubmitHandler } = useSearchValue();
+
+  const [searchInAuthor, setSearchInAuthor] = useState(true);
+
+  const handleClose = () => {
+    props.setOpenDialog(false);
+    setSearchInAuthor(false);
+
+    console.log('close: ' + searchInAuthor);
+  };
+
+  const handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInAuthor(e.target.checked);
```

Comment:
This value might need to be added in the SearchProvider, the toggle is not being read by the provider which is why it's not working

Reply:
I tried to store in Search Provider but same